subsys/net/Kconfig

```diff
@@ -81,6 +81,29 @@ config  NETWORKING
 
 if NETWORKING
 
+config NET_HOSTNAME_ENABLE
+	bool "Add hostname to the device"
+	default n
+	help
+	This is used for example in mDNS to respond to <hostname>.local
+	mDNS queries.
+
+config NET_HOSTNAME
+	string "The hostname of this device"
+	default "zephyr"
+	depends on NET_HOSTNAME_ENABLE
+	help
+	The string should be a valid hostname.
+
+config NET_HOSTNAME_UNIQUE
+	bool "Make hostname unique"
+	default n
+	depends on NET_HOSTNAME_ENABLE
+	help
+	This will append link address to hostname to create a unique
+	hostname. For example, zephyr00005e005357 could be the hostname
+	if this	setting is enabled.
+
```

Comment:
Maybe creating a Kconfig.hostname would be interesting. 

As well as net_hostname.c, just to move it out of net_core.c which is quite big now.

We may have future additions also like dynamically changing the hostname. (and using such functions through net shell would be interesting as well).